torch/quantization/quantize.py

```diff
@@ -102,8 +102,9 @@ def add_observer(module):
     if hasattr(module, 'qconfig') and module.qconfig is not None and \
        len(module._modules) == 0:
         # observer and hook will be gone after we swap the module
-        module.add_module('observer', module.qconfig.activation())
-        module.register_forward_hook(_observer_forward_hook)
+        if module.training and type(module) not in set(DEFAULT_QAT_SKIP_MODULE_LIST) or module.training is False:
```

Comment:
I think this is overcomplicated. just use`if not (module.training and type(module) == DeQuantStub)`

Reply:
this feels like a bit too magical to be so deep in the code. Why can't we just pass DeQuantStub as a skip list to propagate_qconfig? it already has a skip list

Reply:
Started with that design, with a list being passed to propagate_qconfig for the case of quantization aware training.  

Since we are going to change this to a whitelist (after 1.3), we can make this change later.  We need to land this soon, so lets settle on a preferred way soon. 

My recommendation is to have a list just so that the flow is clear to the end user and have it explicitly passed within prepare_qat calls (like the original version of the diff).